File: fyc-api-rest/src/main/java/fr/esgi/fyc_api_rest/rest/ExerciceRestController.java
package fr.esgi.fyc_api_rest.rest;

import fr.esgi.fyc_api_rest.business.Exercice;
import fr.esgi.fyc_api_rest.dto.exercice.in.ExerciceDTO;
import fr.esgi.fyc_api_rest.mapper.ExerciceMapper;
import fr.esgi.fyc_api_rest.service.ExerciceService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/Exercices")
public class ExerciceRestController {
    private final ExerciceService exerciceService;
    private final ExerciceMapper exerciceMapper;

    public ExerciceRestController(ExerciceService exerciceService, ExerciceMapper exerciceMapper) {
        this.exerciceService = exerciceService;
        this.exerciceMapper = exerciceMapper;
    }

    @PostMapping
    @ResponseStatus(code = HttpStatus.CREATED)
    public Exercice create(@RequestBody ExerciceDTO exerciceDTO){
        return exerciceService.create(exerciceMapper.toEntity(exerciceDTO));
    }

    @GetMapping
    @ResponseStatus(code = HttpStatus.OK)
    public List<Exercice> findAll(){
        return exerciceService.findAll();
    }

    @GetMapping("/{id}")
    @ResponseStatus(code = HttpStatus.OK)
    public Exercice findById(@PathVariable Long id){
        return exerciceService.findById(id);
    }

    @PutMapping("/{id}")
    public Exercice update(@PathVariable Long id, @RequestBody ExerciceDTO exerciceDTO){
        return exerciceService.update(id, exerciceMapper.toEntity(exerciceDTO));
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable Long id){
        exerciceService.delete(id);
    }
}

File: fyc-rest-api-csharp-test/CometeAPI/Rest/ExerciceRestController.cs
using FycRestApi.Business;
using FycRestApi.DTO.Request;
using FycRestApi.DTO.Response;
using FycRestApi.Mapper;
using FycRestApi.Service.Interface;
using Microsoft.AspNetCore.Mvc;

namespace FycRestApi.Rest;

[ApiController]
[Route("api/Exercices")]

public class ExerciceRestController : ControllerBase
{
    private readonly IExerciceService _exerciceService;
    private readonly ExerciceMapper _exerciceMapper;
    public ExerciceRestController(IExerciceService exerciceService, ExerciceMapper exerciceMapper)
    {
        this._exerciceService = exerciceService;
        this._exerciceMapper = exerciceMapper;
    }

    [HttpGet("")]
    public async Task<ActionResult<List<ExerciceResponseDTO>>> index()
    {
        List<Exercice> exercices = await this._exerciceService.findAll();
        return Ok(exercices.Select(exercice => _exerciceMapper.toDTO(exercice)).ToList());
    }

    [HttpPost]
    public async Task<ActionResult<ExerciceResponseDTO>> Create([FromBody] ExerciceRequestDTO requestDTO)
    {
        try
        {
            Exercice exercice = await _exerciceService.save(await _exerciceMapper.toEntity(requestDTO));
            string uri = Url.Action(action: nameof(FindById), new { id = exercice.id });
            return Created(uri, exercice);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<ExerciceResponseDTO>> FindById(int id)
    {
        try
        {
            Exercice exercice = await _exerciceService.findById(id);
            return Ok(_exerciceMapper.toDTO(exercice));
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpPut("{id}")]
    public async Task<ActionResult<ExerciceResponseDTO>> update(int id, [FromBody] ExerciceRequestDTO requestDTO)
    {
        try
        {
            Exercice exercice = await _exerciceService.update(id, await _exerciceMapper.toEntity(requestDTO));
            return Ok(_exerciceMapper.toDTO(exercice));
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> delete(int id)
    {
        try
        {
            await _exerciceService.delete(id);
            return NoContent();
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }
}